apps/src/fish/FishView.jsx

```diff
@@ -53,23 +47,98 @@ class FishView extends React.Component {
     mobilePortraitWidth: PropTypes.number.isRequired
   };
 
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      windowWidth: $(window).width(),
+      windowHeight: $(window).height(),
+      appWidth: $('#codeApp').width(),
+      appHeight: $('#codeApp').height()
+    };
+  }
+
   componentDidMount() {
     this.props.onMount();
+
+    this.setState({
+      windowWidth: $(window).width(),
+      windowHeight: $(window).height(),
+      appWidth: $('#codeApp').width(),
+      appHeight: $('#codeApp').height()
+    });
+
+    window.addEventListener('resize', _.debounce(this.onResize, 100));
   }
 
+  onResize = () => {
+    const windowWidth = $(window).width();
+    const windowHeight = $(window).height();
+
+    // Check that the window dimensions have actually changed to avoid
+    // unnecessary event-proessing on iOS Safari.
+    if (
+      this.state.windowWidth !== windowWidth ||
+      this.state.windowHeight !== windowHeight
+    ) {
+      const appWidth = $('#codeApp').width();
+      const appHeight = $('#codeApp').height();
+
+      this.setState({windowWidth, windowHeight, appWidth, appHeight});
+    }
+  };
+
   render() {
+    // The tutorial has a width:height ratio of 16:9.
+    const aspectRatio = 16 / 9;
+
+    let containerWidth, containerHeight;
+
+    // Constrain tutorial to 1280px maximum width.
+    const maxContainerWidth = Math.min(this.state.appWidth, 1280);
+
+    // Reducing by 27px leaves appropriate space above the small footer.
```

Comment:
The small footer is currently this size (including a 5px top margin).

Since you can't derive this easily (it is in a variety of CSS styles from the dashboard application.css), it might be better to just call this the `BOTTOM_MARGIN`